### ISSUE TITLE
Error basic::test::Memory 1

### ISSUE THREAD

**Issue by AnantaYudica** [AnantaYudica/basic#46]
```
#define USING_BASIC_TEST_MEMORY
#include "Test.h"

#include <vector>

struct BaseTest
{
    virtual ~BaseTest() {};
    virtual void Test() = 0;
};

struct TestRegister
{
    static std::vector<BaseTest*> List;
    BaseTest* m_ptr;
    TestRegister(BaseTest* ptr) :
        m_ptr(ptr)
    {
        List.push_back(ptr);
    }
    ~TestRegister()
    {
        delete m_ptr;
    }
};

std::vector<BaseTest*> TestRegister::List;

struct Test1 : BaseTest
{
    void Test()
    {}
};

TestRegister t(new Test1);

int main()
{
    
}

```
`assertion "m_blocks != NULL" failed: file "include/test/Memory.h", line 121, function: void basic::test::Memory<Tout>::Register(void*, std::size_t) [with Tout = basic::test::Output<>; std::size_t = long unsigned int]`

**Comment by AnantaYudica** [comment]
Fixed in v0.1-beta-2.
Please update test source :
```
#define USING_BASIC_TEST_MEMORY
#include "Test.h"

BasicTest& bt = BasicTest::ms_instance;
```

**Comment by AnantaYudica** [comment]
Reopen, not fixed in v0.1-beta-2.

**Comment by AnantaYudica** [comment]
Fixed in v0.1-beta-3.

**Comment by AnantaYudica** [comment]
Update from
`BasicTestConstruct;` to `BASIC_TEST_CONSTRUCT;`
